### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by rtonata88**
In this PR I have refactored the code to following the SOLID OOP principles.

### CODE REVIEW

**Review by julie-ify — CHANGES_REQUESTED** [review]
Hi @rtonata88, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has failed linter. Kindly fix the linter and submit it again

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    when 4 # Create a book
+      create_book
+    when 5 # Create a rental
+      add_rental
+    when 6 # List all rentals a given person ID
+      list_rentals_by_person
+    else # Exit
+      puts 'Goodbye :-)'
+    end
+  end
```
Comment (julie-ify) [inline app.rb]: - [ ] Kindly move everything about starting the app, the code that asks the user for the initial choice of options (or to quit) to the `main.rb` file. This is to achieve the `Single-responsibility [Principle](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid)`

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    end
+    puts "\n \n"
+
+    # Display app main menu
+    start
+  end
+
+  # 3. Create a person
+  def create_person
+    puts 'Do you want to create a (1) student or a (2) teacher [input the number]: '
```
Comment (julie-ify) [inline app.rb]: - [x] All requests for input and input validation should've been placed in the `main.rb` file and/or classes/methods whose responsibilities are to request or validate input.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @rtonata88,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikethreels — APPROVED** [review]
Hi Richard, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**